src/nvm/file.rs

```diff
@@ -115,6 +115,90 @@ impl FileNvmBuilder {
         self
     }
 
+    #[cfg(target_os = "linux")]
+    fn file_open_with_error_info<P: AsRef<Path>>(
+        &self,
+        do_create: bool, // This denoes whether or not the next argument `options` allows file creating.
+        options: &fs::OpenOptions,
+        filepath: &P,
+    ) -> Result<File> {
+        use std::os::unix::fs::OpenOptionsExt;
+
+        let file_orig = track_io!(options.open(&filepath));
+
+        // If we succeed on opening the file `filepath`, we return it.
+        if file_orig.is_ok() {
+            return file_orig;
+        }
+
+        // Otherwise, we check why opening the file `filepath` failed.
+        // First, we check the existence of the file `filepath`.
+        if !std::path::Path::new(filepath.as_ref()).exists() {
+            if do_create {
+                // failed to file open
+                return track!(
```

Comment:
This expression can be written as follows:

```rust
return track!(file_orig, "[CannyLS Error] We failed to create the file {:?}.", filepath.as_ref());
```

And I think the header "[CanyLS Error]" is redundant (crate name is included in a `trackable` history).